storm-core/src/jvm/org/apache/storm/metrics2/reporters/ScheduledStormReporter.java

```diff
@@ -0,0 +1,90 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.metrics2.reporters;
+
+import com.codahale.metrics.ScheduledReporter;
+import org.apache.storm.metrics2.Metrics2Utils;
+import org.apache.storm.metrics2.filters.StormMetricsFilter;
+import org.apache.storm.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public abstract class ScheduledStormReporter implements StormReporter{
+    private static final Logger LOG = LoggerFactory.getLogger(ScheduledStormReporter.class);
+    protected ScheduledReporter reporter;
+    protected long reportingPeriod;
+    protected TimeUnit reportingPeriodUnit;
+
+    @Override
+    public void start() {
+        if (reporter != null) {
+            LOG.debug("Starting...");
+            reporter.start(reportingPeriod, reportingPeriodUnit);
+        } else {
+            throw new IllegalStateException("Attempt to start without preparing " + getClass().getSimpleName());
+        }
+    }
+
+    @Override
+    public void stop() {
+        if (reporter != null) {
+            LOG.debug("Stopping...");
+            reporter.stop();
+        } else {
+            throw new IllegalStateException("Attempt to stop without preparing " + getClass().getSimpleName());
+        }
+    }
+
+
+    public static TimeUnit getReportPeriodUnit(Map<String, Object> reporterConf) {
+        TimeUnit unit = getTimeUnitForConfig(reporterConf, REPORT_PERIOD_UNITS);
+        return unit == null ? TimeUnit.SECONDS : unit;
+    }
+
+    private static TimeUnit getTimeUnitForConfig(Map reporterConf, String configName) {
+        String rateUnitString = Utils.getString(reporterConf.get(configName), null);
+        if (rateUnitString != null) {
+            return TimeUnit.valueOf(rateUnitString);
+        }
+        return null;
+    }
+
+    public static long getReportPeriod(Map reporterConf) {
+        return Utils.getInt(reporterConf.get(REPORT_PERIOD), 10).longValue();
+    }
+
+    public static StormMetricsFilter getMetricsFilter(Map reporterConf){
+        StormMetricsFilter filter = null;
+        Map<String, Object> filterConf = (Map)reporterConf.get("filter");
+        if(filterConf != null) {
+            String clazz = (String) filterConf.get("class");
+            if (clazz != null) {
+                try {
+                    filter = (StormMetricsFilter) Metrics2Utils.instantiate(clazz);
+                    filter.prepare(filterConf);
+                } catch (Exception e) {
+                    LOG.warn("Unable to instantiate StormMetricsFilter class: {}", clazz);
```

Comment:
Same question as the other use of Metrics2Utils, why don't we want to crash?

Reply:
I could go either way... But it didn't seem right to crash just because of a misconfigured filter.

Reply:
Sorry, forgot to respond to this. Don't we usually crash if Storm is misconfigured in some way? I'm worried it will be less visible to users if we quietly(-ish, there's a warning in the log) do the wrong thing, rather than crashing.

Reply:
Okay. I changed it to crash instead.